console-ui/src/main/java/org/jline/consoleui/prompt/ConsolePrompt.java

```diff
@@ -179,39 +173,47 @@ public void prompt(
 
         for (int i = resultMap.isEmpty() ? 0 : resultMap.size() - 1; i < promptableElementList.size(); i++) {
             PromptableElementIF pe = promptableElementList.get(i);
-            PromptResultItemIF result = promptElement(header, pe);
-            if (result == null) {
-                // Prompt was cancelled by the user
-                if (i > 0) {
-                    // Remove last result
-                    header.remove(header.size() - 1);
-                    // Go back to previous prompt
-                    i -= 2;
-                    continue;
-                } else {
-                    if (config.cancellableFirstPrompt()) {
+            try {
```

Comment:
Any reason to prefer the inner loop rather than the outer scope ? just wondering...

Reply:
Does it make any difference? Loop with be broken anyway. Also, this way "catch is closer to the source", no?